docs/tutorials/build_a_node_in_rust.md

```diff
@@ -691,9 +688,46 @@ fn handle_ldk_event(..) {
 
 **Dependencies:** `ChannelManager`, `ChainMonitor`, `KeysManager`, `BroadcasterInterface`
 
-**References:** [`Event` docs](https://docs.rs/lightning/*/lightning/util/events/enum.Event.html), [LDK sample node event handling example](https://github.com/lightningdevkit/ldk-sample/blob/bc07db6ca4a3323d8718a27f85182b8157a20750/src/main.rs#L101-L240), [`EventHandler` docs](https://docs.rs/lightning/*/lightning/util/events/trait.EventHandler.html)
+**References:** [`Event` docs](https://docs.rs/lightning/*/lightning/util/events/enum.Event.html), [`EventHandler` docs](https://docs.rs/lightning/*/lightning/util/events/trait.EventHandler.html),[LDK sample node event handling example](https://github.com/lightningdevkit/ldk-sample/blob/bc07db6ca4a3323d8718a27f85182b8157a20750/src/main.rs#L101-L240)
+
+### Step 16. Initialize the `ProbabilisticScorer` 
+**What it's used for:** to find a suitable payment path to reach the destination.
+
+```rust
+let params = ProbabilisticScoringParameters::default()
+
+let scorer = ProbabilisticScorer::new(params, &network_graph)
+```
+
+**Dependencies** `NetworkGraph`
+
+**Reference** [`ProbabilisticScorer` docs](https://docs.rs/lightning/*/lightning/routing/scoring/type.ProbabilisticScorer.html),
+[LDK sample node ProbabilisticScorer example](https://github.com/lightningdevkit/ldk-sample/blob/main/src/main.rs#L612)
+
+### Step 17. Initialize the `InvoicePayer`
+**What it's used for:** to create an invoice payer that retries failed payment paths.
+
+```rust
+let router = DefaultRouter::new(&network_graph, &logger, keys_manager.get_secure_random_bytes())
+
+let invoice_payer = InvoicePayer::new(
+	&channel_manager, 
+	router,
+	&scorer,
+	&logger,
+	event_handler
+	payment::RetryAttempts(5)
+)
+```
+
+**Implementation notes:** The scorer is used when finding a route and when getting events from successful and failed paths. The retrier consumes those events by telling the scorer that a path has failed or succeeded and then retries by querying for a (potentially new) path with the updated scorer. LDK users can use the LDK scorer and their own retry logic, the latter of which delegates to the scorer.
```

Comment:
Let's drop the last sentence since the `InvoicePayer` contains all the retry logic and a custom retry requires a it more explanation.